telescope/servicecontrol.cpp

```diff
@@ -0,0 +1,117 @@
+#include "servicecontrol.h"
+
+#include <QFile>
+#include <QDir>
+#include <QDebug>
+#include <QCoreApplication>
+#include <QProcess>
+
+ServiceControl::ServiceControl(QObject *parent) : QObject(parent)
+{
+
+}
+
+QString ServiceControl::serviceName() const
+{
+    return m_serviceName;
+}
+
+void ServiceControl::setServiceName(const QString &serviceName)
+{
+    if (m_serviceName != serviceName) {
+        m_serviceName = serviceName;
+        emit serviceNameChanged();
+    }
+}
+
+bool ServiceControl::serviceFileInstalled() const
+{
+    if (m_serviceName.isEmpty()) {
+        qDebug() << "Service name not set.";
+        return false;
+    }
+    QFile f(QDir::homePath() + "/.config/upstart/" + m_serviceName + ".conf");
+    return f.exists();
+}
+
+bool ServiceControl::installServiceFile()
```

Comment:
Do we really need that kind of hack? Couldn't we write that service file ourselves?